tests/Fixtures/TestBundle/GraphQl/Type/Definition/UploadType.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Tests\Fixtures\TestBundle\GraphQl\Type\Definition;
+
+use ApiPlatform\Core\GraphQl\Type\Definition\TypeInterface;
+use GraphQL\Error\Error;
+use GraphQL\Error\InvariantViolation;
+use GraphQL\Type\Definition\ScalarType;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+/**
+ * Represents an Upload type.
+ *
+ * @author Mahmood Bazdar <mahmood@bazdar.me>
+ */
+final class UploadType extends ScalarType implements TypeInterface
```

Comment:
Maybe we should provide this type directly :thinking:

Reply:
I agree, I think it helps users. because they need to define it if they want to add upload request.
 If your decision is final, I can move it to `src/GraphQl/Type/Definition`.